comment_scan.rb

```diff
@@ -440,6 +440,30 @@ def restart(num_to_post, bundle)
         say reasons.join("\n")
       end
     end
+    command "!!/regexstats" do |bot, reason|
+      if matches_bot(bot)
+        #Build array of hashes for each regex containing info to build the stat output
+        regexes = (reason.nil? ? Reason.all : Reason.where(name: reason)).map do |r|
+          r.regexes.map do |regex| 
+            tps = Comment.where("tps >= ?", 1).count { |comment| %r{#{regex.regex}}.match(comment.body_markdown.downcase) }
+            fps = Comment.where("fps >= ?", 1).count { |comment| %r{#{regex.regex}}.match(comment.body_markdown.downcase) }
+            {"effectivePercent" => (tps + fps > 0) ? tps/(tps + fps.to_f) : 0, 
+             "tps" => tps, "fps" => fps, "postType" => regex.post_type, "regex" => regex.regex, "reason" => r.name}
+          end
+        end
+        regexes = regexes.flatten.sort_by { |regex| regex["effectivePercent"] } #Order by effectiveness
```

Comment:
Generally, prefer symbol keys to string keys in the hash. Symbols are neater than strings for things which are immutable.